packages/typespec-vscode/test/extension/common/commonSteps.ts

```diff
@@ -0,0 +1,355 @@
+      `${command == "Emit from TypeSpec" ? "emit" : "preview"}_typespec.png`)
+    })
+  } else if (command == "Import TypeSpec from Openapi 3") {
+    const targetName =
+      type === "emptyfolder"
+        ? "ImportTypespecProjectEmptyFolder"
+        : "openapi.3.0.yaml"
+    const target = page.getByRole("treeitem", { name: targetName }).locator("a")
+    await target.click({ button: "right" })
+    const __dirname = path.dirname(fileURLToPath(import.meta.url));
+    await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/openapi.3.0.png") })
+    await sleep(3)
+    await page
+      .getByRole("menuitem", { name: "Import TypeSpec from OpenAPI" })
+      .click()
+    await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/import_typespec.png") })
+  }
+}
+
+/**
+ * In vscode, when you need to select a folder or a file, call this method
+ * @param file When selecting a file, just pass it in. If you need to select a folder, you do not need to pass this parameter in.
+ */
+async function selectFolder(page: Page, file: string = "") {
+  await sleep(10)
+  await keyboard.type(file)
+  const __dirname = path.dirname(fileURLToPath(import.meta.url));
+  await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/select_folder.png") })
+  await keyboard.pressKey(Key.Enter)
+  await keyboard.releaseKey(Key.Enter)
+  await sleep(3)
+}
+
+/**
+ * If the current folder is not empty, sometimes a pop-up will appear
+ * asking "Do you want to continue selecting the current folder as the root directory?".
+ * In this method, select "yes" because selecting "no" does not make sense.
+ * @param page vscode object
+ */
+async function notEmptyFolderContinue(page: Page) {
+  let yesBtn: Locator
+  await retry(
+    page,
+    5,
+    async () => {
+      yesBtn = page.getByRole("option", { name: "Yes" }).locator('label')
+        .filter({ hasText: "Yes" }).first()
+      let noBtn = page.getByRole("option", { name: "No" }).locator('label')
+        .filter({ hasText: "No" }).first() 
+      return (await yesBtn.count() > 0) && (await noBtn.count() > 0) 
+    },
+    "Failed to find yes/no button",
+    1
+  )
+  await retry(
+    page,
+    5,
+    async () => {
+      let yesdescriptionBox = page.getByRole("option", { name: "Yes" }).locator('label')
+      let yesdescriptionText = await yesdescriptionBox.textContent();
+      return yesdescriptionText !== null && (yesdescriptionText.includes("YesSelected folder"))
+    },
+    "Failed to match the description for the non-empty folder cases",
+    1
+  )
+  const __dirname = path.dirname(fileURLToPath(import.meta.url));
+  await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/not_empty_folder_continue.png") })
+  await yesBtn!.click()
+}
+
+/**
+ * Install plugins directly from vscode
+ * @param page vscode object
+ */
+async function installExtension(page: Page) {
```

Comment:
is this used? feels liek thats not really the point of testing the extension anyway